### PR TITLE
Migrate the tests in main-traversal.cxx to GTest

### PR DESCRIPTION
**Pull request by Dannnno**
Hi, I'm a collaborator with the Harvey Mudd clinic team and we had rewritten these tests in advance of some of our work involving Agency.

- Enhances #50 and #89 
- Moved tests from main-traversal.cxx to GTest format
- Tests of index set equality already existed in test-indexsets.cxx
- Conditional index set operations tests moved to test-indexsets.cxx
- All forall and forall_Icount tests moved to test-forall.cxx
- Added test-forall.cxx to GTest test runner

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
Sweet, initial once over looks good, but I'll leave it to our testing experts ( @davidbeckingsale doesn't seem to be running quickly enough these days) for a final review.

Thanks for the contribution!

File: test/unit/cpu/main-traversal.cxx [hunk 1]
```diff
@@ -1,553 +0,0 @@
...
-    last_indx =
-        max(last_indx,
-            buildIndexSet(index, static_cast<IndexSetBuildMethod>(ibuild)));
-#if 0  // print index set for debugging
-      cout << "\n\nIndexSet( " << ibuild << " ) " << endl;
-      index[ibuild].print(cout);
-#endif
-  }
-
-  ///////////////////////////////////////////////////////////////////////////
```
Comment (Dannnno) [inline test/unit/cpu/main-traversal.cxx]: This part appears to already exist in test-indexsets.cxx

File: test/unit/cpu/main-traversal.cxx [hunk 2]
```diff
@@ -1,553 +0,0 @@
...
-      cout << endl << endl;
-#endif
-  }
-
-  cout << "\n tests passed / test run: " << s_ntests_passed << " / "
-       << s_ntests_run << endl;
-
-  cout << "\n END IndexSet equality/inequality tests " << endl;
-
-  ///////////////////////////////////////////////////////////////////////////
```
Comment (Dannnno) [inline test/unit/cpu/main-traversal.cxx]: Also exists in test-indexsets.cxx

File: test/unit/cpu/main-traversal.cxx [hunk 3]
```diff
@@ -1,553 +0,0 @@
...
-  }
-
-  cout << "\n tests passed / test run: " << s_ntests_passed << " / "
-       << s_ntests_run << endl;
-
-  cout << "\n END IndexSet invalid segment tests " << endl;
-
-  ///////////////////////////////////////////////////////////////////////////
-  //
-  // Run RAJA::forall loop iteration tests...
```
Comment (Dannnno) [inline test/unit/cpu/main-traversal.cxx]: These were moved into test-forall.cxx

File: test/unit/cpu/main-traversal.cxx [hunk 4]
```diff
@@ -1,553 +0,0 @@
...
-  for (unsigned ibuild = 0; ibuild < run_tests; ++ibuild) {
-    runForall_IcountTests(
-        ibuild, parent, array_length, index[ibuild], is_indices);
-  }
-
-#if !defined(RAJA_COMPILER_XLC12) && 1
-
-  ///////////////////////////////////////////////////////////////////////////
-  //
-  // Check some basic conditional IndexSet operations....
```
Comment (Dannnno) [inline test/unit/cpu/main-traversal.cxx]: This was moved into test-indexsets.cxx

**Pull request merged**